src/sxmlc.c

```diff
@@ -2178,7 +2187,7 @@ int read_line_alloc(void* in, DataSourceType in_type, SXML_CHAR** line, int* sz_
 			} else
 				*line = pt;
 		}
-		if (sz_line && n < *sz_line)
+		if (line && sz_line && n < *sz_line)

```

Comment:
Which version of sxmlc are you using? In the latest (4.5.1), `line` cannot be NULL here.

Reply:
Indeed, and I understand that you're puzzled. I added that because Visual Studio was claiming a Buffer Overrun on this line. All in all this resolved nothing, I just forgotten to remove it, you can thus get rid of that. The warning I receive is the following:
Warning	C6386	Buffer overrun while writing to '*line':  the writable size is '*sz_line*sizeof(SXML_CHAR)' bytes, but '2' bytes might be written. ...\sxmlc.c	2191

Reply:
NOTE I'm compiling in 64 bit, it's may be the reason why I receive many more warnings than other users ;-)